### PR TITLE
Closes #102: Allow interactive exploration of datasets in vignette

### LINKED ISSUE #102: Documentation:  Interactively explore datasets in referrence pages/vignette
### Please select a category the issue is focused on?

_No response_

### Let us know where something needs a refresh or put your idea here!

It would be nice to be able to peruse each dataset within a vignette or the reference page. Open to ideas on how to achieve this!

### PR DESCRIPTION
**Pull request by Gero1999**
> Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

> Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

___

## Implementation description

Closes #102 

* Added a new vignette `preview-adam.Rmd` that displays all datasets in the package in collapsible panels with filtering and preview options. This allows users to interactively browse and filter up to the first 50 rows of each dataset.
* Updated the package website navigation (`_pkgdown.yml`) to add a "Explore ADAM Datasets" link in the navbar, pointing to the new vignette.
* Included vignette (or any article) in .Rbuildignore

___

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build pharmaverseadam site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new datasets occur on the "[Reference](https://pharmaverse.github.io/pharmaverseadam/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side. >> `cannot do it myself` Closes #102 

- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: vignettes/articles/preview-adam.Rmd [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+```{r echo=FALSE}
+library(pharmaverseadam)
+library(htmltools)
+library(reactable)
+
+dataset_summary <- data(package = "pharmaverseadam")
+dataset_names <- dataset_summary$results[, "Item"]
+dataset_titles <- dataset_summary$results[, "Title"]
+
```
Comment (Fanny-Gautier) [inline vignettes/articles/preview-adam.Rmd]: Please sort the datasets per Therapeutic Area (see issue PR #140 ).

**Review by Fanny-Gautier — CHANGES_REQUESTED** [review]
It's looking great! Thank you for implementing it to `{pharmaverseadam}`!

**Review by Fanny-Gautier — CHANGES_REQUESTED** [review]
Implemented updates are good! Remaining implementation is the organization by Therapeutic Area. Thank you.

File: vignettes/articles/preview-adam.Rmd (hunk 1 above)
Comment (Gero1999) [reply vignettes/articles/preview-adam.Rmd]: I see, I changed the code to use the specs and integrate this logic to derive therapeutic area from them as well as the in-package dataset names ("_ophtha", "_vaccine", "_peds"...). 

I also included a check that will stop the website rendering if any package was not created using consistently this terminology. It will share an error message specifying which dataset has conflicting terminology. I guessed it was a good idea, but maybe is not. Let me know what you think though!

**Review by Fanny-Gautier — CHANGES_REQUESTED** [review]
Could you also add a statement in the `README.md`, for example within the `Generates Dataset Documentation` section, noting that the data exploration is generated via `vignettes/articles/preview-adam.Rmd` and that if a therapeutic area is added to the package, the vignette must also be updated to include this new therapeutic area?
Thank you.

Edit: or in `3. Generates Dataset Exploration` section since it is outside of `create_adams_data.R`. Up to you.

**Comment by Fanny-Gautier** [comment]
@Gero1999 Please tick all the boxes in the checklist, especially the one that says "**Pat yourself on the back for a job well done! Much love to your accomplishment!**" :wink:
